url_shortener_proj/app/Http/Controllers/UrlController.php

```diff
@@ -44,15 +42,42 @@ public function details($detail)
         if (!isset($url)) {
             abort(404);
         }
+
         return view('details')->withUrl($url);
     }
 
-    public function views($id)
+    public function views($short, Request $request)
     {
-        $url = Url::find($id);
+        $url = Url::whereShort($short)->first();
```

Comment:
What will happen if URL with this `$short` doesn't exist in the database?

Reply:
I added abort(404) in case that url doesn't exists.